package NYT.NClickHouseServer.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";
import "yt_proto/yt/core/misc/proto/error.proto";

////////////////////////////////////////////////////////////////////////////////

message TChytRequest
{
    optional string query = 1;
    map<string, string> settings = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqExecuteQuery
{
    optional TChytRequest chyt_request = 1;
    optional int64 row_count_limit = 2;
    optional NYT.NProto.TGuid query_id = 3;
}

message TRspExecuteQuery
{
    optional NYT.NProto.TGuid query_id = 1;
    optional NYT.NProto.TError error = 2;
    // Attachments contain wire-encoded data.
}

////////////////////////////////////////////////////////////////////////////////

message TProgressValues
{
    optional int64 read_rows = 1;
    optional int64 read_bytes = 2;
    optional int64 total_rows_to_read = 3;
    optional int64 total_bytes_to_read = 4;
    optional bool finished = 5;
}

message TQueryProgressValues
{
    optional TProgressValues total_progress = 1;
    repeated NYT.NProto.TGuid secondary_query_ids = 2;
    repeated TProgressValues secondary_query_progresses = 3;
    optional NYT.NProto.TGuid query_id = 4;
}

message TMultiQueryProgressValues
{
    optional int64 queries_count = 1;
    repeated TQueryProgressValues progresses = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqGetQueryProgress
{
    required NYT.NProto.TGuid query_id = 1;
}

message TRspGetQueryProgress
{
    optional TQueryProgressValues progress = 1;
    optional TMultiQueryProgressValues multi_progress = 2;
}

////////////////////////////////////////////////////////////////////////////////
